app/controllers/admin/products_controller.rb

```diff
@@ -61,9 +71,13 @@ def update
 
       def destroy
         begin
-          product = ::Stripe::Product.delete(params[:id])
+          external_product = ::Stripe::Product.delete(params[:id])
 
-          render_json_dump product
+          product = Product.find_by(external_id: params[:id])
+
+          product.delete if product
```

Comment:
Minor thing, but by using `delete_by` you will have one SQL query - https://apidock.com/rails/v6.0.0/ActiveRecord/Relation/delete_by